Step Number Missing From Binary Testing TestCheckFunc Errors

### SDK version

```
v2.0.0
```

### Relevant provider source code

```go
func TestAccAwsBackupPlan_withLifecycle(t *testing.T) {
	var plan backup.GetBackupPlanOutput
	ruleNameMap := map[string]string{}
	resourceName := "aws_backup_plan.test"
	rName := fmt.Sprintf("tf-testacc-backup-%s", acctest.RandStringFromCharSet(14, acctest.CharSetAlphaNum))

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSBackup(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAwsBackupPlanDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccAwsBackupPlanConfig_lifecycleColdStorageAfterOnly(rName),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAwsBackupPlanExists(resourceName, &plan, &ruleNameMap),
					resource.TestCheckResourceAttr(resourceName, "name", rName),
					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "rule_name", rName),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.0.cold_storage_after", "7"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.0.delete_after", "0"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "recovery_point_tags.%", "0"),
				),
			},
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
			{
				Config: testAccAwsBackupPlanConfig_lifecycleDeleteAfterOnly(rName),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAwsBackupPlanExists(resourceName, &plan, &ruleNameMap),
					resource.TestCheckResourceAttr(resourceName, "name", rName),
					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "rule_name", rName),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.0.cold_storage_after", "0"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.0.delete_after", "120"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "recovery_point_tags.%", "0"),
				),
			},
			{
				Config: testAccAwsBackupPlanConfig_lifecycleColdStorageAfterAndDeleteAfter(rName),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAwsBackupPlanExists(resourceName, &plan, &ruleNameMap),
					resource.TestCheckResourceAttr(resourceName, "name", rName),
					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "rule_name", rName),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.0.cold_storage_after", "30"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.0.delete_after", "180"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "recovery_point_tags.%", "0"),
				),
			},
			{
				Config: testAccAwsBackupPlanConfig_basic(rName),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAwsBackupPlanExists(resourceName, &plan, &ruleNameMap),
					resource.TestCheckResourceAttr(resourceName, "name", rName),
					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "rule_name", rName),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.#", "0"),
					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "recovery_point_tags.%", "0"),
				),
			},
		},
	})
}
```

### Terraform Configuration Files

Long and complicated with ☝️ specifically (you can look in the Terraform AWS Provider repository), but any `TestCheckFunc` failure I'm guessing should show this.

### Debug Output

Ask and you shall receive.

### Expected Behavior

Previously, the acceptance testing framework would output the step number, so developers can easily troubleshoot when the check failure occurred. Note the `Step 2 error` below (sorry for using a different test failure here, but hopefully you get the idea).

```
TestAccAWSDBInstance_cloudwatchLogsExportConfigurationUpdate: testing.go:684: Step 2 error: Check failed: Check 3/4 error: aws_db_instance.bar: Attribute 'enabled_cloudwatch_logs_exports.1' expected "general", got "error"
--- FAIL: TestAccAWSDBInstance_cloudwatchLogsExportConfigurationUpdate (649.31s)
```

### Actual Behavior

```
TestAccAwsBackupPlan_withLifecycle: resource_aws_backup_plan_test.go:191: Check 7/8 error: aws_backup_plan.test: Attribute 'rule.0.lifecycle.0.delete_after' not found
--- FAIL: TestAccAwsBackupPlan_withLifecycle (15.22s)
```

Which 7th check failed out of the multiple possible steps?

### Steps to Reproduce

After finding a test with a consistently failing `TestCheckFunc`:

`TF_ACC=1 go test -run='TestAccXXX' -v ./...`

--- a/helper/resource/testing_new.go
+++ b/helper/resource/testing_new.go
@@ -116,14 +116,14 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 			err := testStepNewConfig(t, c, wd, step)
 			if step.ExpectError != nil {
 				if err == nil {
-					t.Fatal("Expected an error but got none")
+					t.Fatal("Step %d, expected an error but got none")
 				}
 				if !step.ExpectError.MatchString(err.Error()) {
-					t.Fatalf("Expected an error with pattern, no match on: %s", err)
+					t.Fatalf("Step %d, expected an error with pattern, no match on: %s", err)
 				}
 			} else {
 				if err != nil {
-					t.Fatal(err)
+					t.Fatalf("Step %d error: %s", err)
 				}
 			}
 			appliedCfg = step.Config

--- a/helper/resource/testing_new.go
+++ b/helper/resource/testing_new.go
@@ -99,7 +99,7 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 				t.Fatal(err)
 			}
 			if skip {
-				log.Printf("[WARN] Skipping step %d", i)
+				log.Printf("[WARN] Skipping step %d/%d", i+1, len(c.Steps))
 				continue
 			}
 		}
@@ -116,14 +116,14 @@ func runNewTest(t testing.T, c TestCase, helper *tftest.Helper) {
 			err := testStepNewConfig(t, c, wd, step)
 			if step.ExpectError != nil {
 				if err == nil {
-					t.Fatal("Step %d, expected an error but got none")
+					t.Fatalf("Step %d/%d, expected an error but got none", i+1, len(c.Steps))
 				}
 				if !step.ExpectError.MatchString(err.Error()) {
-					t.Fatalf("Step %d, expected an error with pattern, no match on: %s", err)
+					t.Fatalf("Step %d/%d, expected an error with pattern, no match on: %s", i+1, len(c.Steps), err)
 				}
 			} else {
 				if err != nil {
-					t.Fatalf("Step %d error: %s", err)
+					t.Fatalf("Step %d/%d error: %s", i+1, len(c.Steps), err)
 				}
 			}
 			appliedCfg = step.Config

